Functions/xASL_quant_Basil.m

```diff
@@ -123,170 +130,203 @@
 
 
 
-function [BasilOptions] = xASL_quant_Basil_Options(pathBasilOptions, x, PWI, bMultiPLD)
+function [BasilOptions] = xASL_quant_Basil_Options(pathBasilOptions, x, PWI)
 % Save a Basil options file and store CLI options for Basil
+% 1. Create an option file
+% 2. Basic tissue parameters
+% 3. Basic acquisition parameters
+% 4. Model fiting parameters
+% 5. Extra features on demand
+% 6. Save BASIL options file
+
+%% 1. Create option_file that contains options which are passed to Fabber
+% basil_options is a character array containing CLI args for the Basil command
+
+FIDoptionFile = fopen(pathBasilOptions, 'w+');
+BasilOptions = '';
+
+fprintf(FIDoptionFile, '# Basil options written by ExploreASL\n');
+fprintf(FIDoptionFile, '--iaf=diff\n'); % as input is PWI
+
+%% 2. Basic tissue parameters
+fprintf(FIDoptionFile, '--t1b=%f\n', x.Q.BloodT1/1000);
+fprintf(FIDoptionFile, '--t1=%f\n', x.Q.TissueT1/1000);
+
+%% 3. Basic acquisition parameters
+
+% Labelling type - PASL or pCASL
+switch lower(x.Q.LabelingType)
+	case 'pasl'
+		fprintf('BASIL: PASL model\n');
+		TIs = (unique(x.Q.Initial_PLD))'/1000;
+
+		% Print all the TIs
+		if x.modules.asl.bMultiPLD
+			% For Time-encoded, we skip the first volume
+			if x.modules.asl.bTimeEncoded
+				TIs = TIs(2:end);
+			end
+			for iTI = 1:length(TIs)
+				fprintf(FIDoptionFile, '--ti%d=%.2f\n', iTI, TIs(iTI));
+			end
+			
+		else
+			fprintf(FIDoptionFile, '--ti=%.2f\n', TIs);
+		end
+		
+		% Either print bolus duration or unspecify it
+		if isfield(x.Q,'LabelingDuration') && x.Q.LabelingDuration
+			fprintf(FIDoptionFile, '--tau=%.2f\n', x.Q.LabelingDuration/1000);
+		else
+			% Bolus duration not know. If multi-TI, then try to infer it
+			if length(TIs) > 1
+				fprintf(FIDoptionFile, '--infertau\n');
+				fprintf('BASIL: Infer bolus duration component\n')
+			end
+		end
+		
+	case {'casl','pcasl'}
+		fprintf(FIDoptionFile, '--casl\n');
```

Comment:
Although this is fine, we should add `--pasl` to the PASL case to make it clear

Reply:
Good point. But PASL is the default option and it doesn't need to be added to the config file. Besides, that option doesn't exist ;)
I added some explanatory comments to the code here: 4dffd3ca